### PR TITLE
Enable Travis CI tests for Swift, ObjC clients and move some tests to CircleCI

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

- use MacOS instead of Ubuntu so as to test Swift and ObjC API clients
- move some tests to CircleCI

After this PR, we'll update https://github.com/swagger-api/swagger-codegen/wiki/Integration-Tests with more details about what tests are run by the CI providers.

### CODE REVIEW

**Comment by wing328** [comment]
There are 6 tests related to swift and swift3 clients. Not all of them passed the Travis CI tests so I'll reply with more details on the failed tests once the test result is ready:

cc @jgavris @jaz-ah @Edubits

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 sudo: required
-language: java
-jdk:
-  - oraclejdk7
-  - oraclejdk8
-
+language: objective-c
+osx_image: xcode7.3
```
Comment (jgavris) [inline .travis.yml]: We probably want Xcode 8.3 right?

Comment (wing328) [inline .travis.yml]: @jgavris sure, I'll update that.

Comment (jgavris) [inline .travis.yml]: I don't think Swift 3 was in Xcode 7.3 so you will probably have to.

**Comment by jgavris** [comment]
Perhaps something to consider after the tests are passing, but the output is so verbose I would expect most people will have a hard time even finding their failures.

**Comment by wing328** [comment]
> Perhaps something to consider after the tests are passing, but the output is so verbose I would expect most people will have a hard time even finding their failures.

Let me try with `xcpretty`

**Comment by jgavris** [comment]
There's that, but swagger-codegen itself also is quite talkative :)

**Comment by wing328** [comment]
We are already using `-q` (quiet mode) in maven and will see what else we can do to further reduce the noise in the output.
```
mvn -q --batch-mode verify -Psamples
```

**Comment by wing328** [comment]
UPDATE: When I ran the integration test for Swift3 PromiseKit locally in my machine, I got the following errors:
```

Testing failed:
	Overriding instance method must be as accessible as its enclosing type
** TEST FAILED **


The following build commands failed:
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
	CompileSwift normal x86_64 /Users/williamcheng/Code/swagger-api/swagger-codegen/samples/client/petstore/swift3/promisekit/SwaggerClientTests/Pods/PromiseKit/Extensions/Foundation/Sources/NSObject+Promise.swift
(2 failures)
```

Command to reproduce the issue:
```
cd samples/client/petstore/swift3/promisekit/SwaggerClientTests
mvn integration-test
```

File: .travis.yml (hunk 1 above)
Comment (jgavris) [inline .travis.yml]: Sorry my bad, Xcode 8.3 actually removed support for Swift 2.3. You should be safe on 8.2?

Comment (wing328) [inline .travis.yml]: Np. I'll update with 8.2 instead.

Comment (wing328) [inline .travis.yml]: Updated with 8.2 but still got the same error message shown below.

**Comment by jgavris** [comment]
@wing328 Try bumping promisekit in the podspec? https://github.com/swagger-api/swagger-codegen/commit/87bd1b7b924d588dfa38f28334d32972b9fa6936

**Comment by jgavris** [comment]
https://github.com/mxcl/PromiseKit/issues/575

**Comment by wing328** [comment]
UPDATE: all Swift (2.x, 3.x) pass but there's an issue running Petstore server locally (Docker service is not supported in the MacOS build image). I'll look into it more this week.

I did 2 fixes to the Swift client or test case

https://github.com/swagger-api/swagger-codegen/pull/5687/commits/819533a3dc0ee5b852832682e02a43eca6d32a01
https://github.com/swagger-api/swagger-codegen/pull/5687/commits/34fa8096f0f6c3e49164ed89f712bab7b261e048

Please review as I'm not sure if my fixes are entirely correct.

**Comment by jgavris** [comment]
🎉 awesome thanks @wing328 !

**Comment by jaz-ah** [comment]
+1 nice job guys!

**Comment by wing328** [comment]
FYI. ObjC tests added via d6743b4597730ccd75ae3514aad0e41ef2ca50f1

**Pull request merged**